### PR TITLE
Add deploy script and improve the dev script

### PR DESCRIPTION
**Pull request by kaiomagalhaes**

### CODE REVIEW

**Comment by kaiomagalhaes** [comment]
As soon as the merge is made I will send another with the url fixes

**Comment by kaiomagalhaes** [comment]
@halissonvit @codelitt @Lockyy

File: best_practices/deploy/deploy [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#############MANDATORY#########
+PROJECT_NAME=
+###############################
+
+if [ -z "$DOCKERHUB_USER" ]; then
+  DOCKERHUB_USER='codelittinc'
+fi
+
+if [ -z "$DEPLOY_TYPE" ]; then
+  exit "You need to pass the deploy type(DEPLOY_TYPE), example: qa, production, beta e etc."
```
Comment (codelitt) [inline best_practices/deploy/deploy]: `exit "You need to pass the deploy type(DEPLOY_TYPE), example: qa, production, beta e etc."`

Should not have extra "e" before "etc". Just a comma -> ","

File: best_practices/deploy/deploy [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+fi
+
+if [ -z "$SERVER_USER" ]; then
+  exit "You need to pass the server user(SERVER_USER) of the application will be running."
```
Comment (codelitt) [inline best_practices/deploy/deploy]: Grammar correction: 

`exit "You need to pass the server username (SERVER_USER) for the deployment of this application."`

File: best_practices/deploy/deploy [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+fi
+
+if [ -z "$SERVER_HOST" ]; then
+  exit "You need to pass the server host(SERVER_HOST) of the application will be running."
```
Comment (codelitt) [inline best_practices/deploy/deploy]: @kaiomagalhaes Grammar correction:

`exit "You need to pass the server hostname (SERVER_HOST) for the deployment of this application."`

File: best_practices/environment.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+[environment.sh](https://github.com/kaiomagalhaes/incubator-resources/blob/master/best_practices/environments/environment.sh)
+
+## 2. Dockerfiles used by above script 
+
+### Dockerfile in your repo
+
+This is the Dockerfile.development file that is called by the shell script above. The Dockerfile.production is similar.
+
+[Dockerfile.development](https://github.com/kaiomagalhaes/incubator-resources/blob/master/best_practices/environments/Dockerfile.development)
```
Comment (codelitt) [inline best_practices/environment.md]: This link will need to be updated to the codelittinc/incubator-resources. Currently links to Kaio's fork

File: best_practices/environment.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+=======
+
+We've fairly scripted method for maintaining environments across teams, staging, and production. The process/scripts below are for a standard Ruby/React stack, but these can be customized to fit your needs.
+
+## 1. The bash script that starts it all
+This is a shell script which basically any engineer can run in development or production to spin everything up. 
+It will live in `bin/dev.sh`.
+It handles the build of the docker images and containers. You can customize it to your liking and environment.
+
+[environment.sh](https://github.com/kaiomagalhaes/incubator-resources/blob/master/best_practices/environments/environment.sh)
```
Comment (codelitt) [inline best_practices/environment.md]: Needs to link to codelittinc fork, not Kaio's

File: best_practices/environment.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+### Dockerfile in your repo
+
+This is the Dockerfile.development file that is called by the shell script above. The Dockerfile.production is similar.
+
+[Dockerfile.development](https://github.com/kaiomagalhaes/incubator-resources/blob/master/best_practices/environments/Dockerfile.development)
+
+### Base image called from Dockerfile.development
+
+[Image](https://github.com/kaiomagalhaes/dockerfiles/blob/master/ruby/Dockerfile)
```
Comment (codelitt) [inline best_practices/environment.md]: Same. Links to wrong repo .

File: best_practices/environment.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+#### This uses Codelitt's Ruby image found at
+
+[Ruby Dockerfile](https://github.com/kaiomagalhaes/dockerfiles/blob/master/ruby/Dockerfile)
```
Comment (codelitt) [inline best_practices/environment.md]: Same. Links to kaio's repo.

File: best_practices/environment.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+## 3. The makefile for building assets
+
+This makefile predefines several commands which build our assets and our frontend app (React) and is largely used with webpack. It lives alongside the Dockerfile.development and Dockerfile.production in the root directory of the repo.  
+
+[Makefile](https://github.com/kaiomagalhaes/incubator-resources/blob/master/best_practices/environments/Makefile)
```
Comment (codelitt) [inline best_practices/environment.md]: Same. Links to kaio's repo. Should not.

**Comment by codelitt** [comment]
@kaiomagalhaes Fix the grammar stuff. Merge. Then fix the URL stuff. 

Other than that, :+1:. Looks good! :D

**Pull request merged**